### DISCUSSION TITLE
Problems with Reloading on Vercel

### DISCUSSION THREAD

**Discussion by TheRealSalmon** [leptos-rs/leptos#2629] [Q&A] [answered]
I deployed my leptos webpage (csr) to vercel but I get a weird error. When I navigate to a different route, like https://sam-mun.vercel.app/projects then reload, I get a 404 error. I tried playing with `vercel.json` but had no luck. Has anyone solved this problem before?

Here is my webpage, https://sam-mun.vercel.app/

**Comment by gbj** [comment, marked as answer]
Generally speaking, if you are deploying any single-page app with client-side routing to any static server, you need to set up some way to rewrite any non-API requests to `index.html` so that they serve the HTML page that will load your client-side app. Glancing at Vercel docs it looks like [`rewrites`](https://vercel.com/docs/projects/project-configuration#rewrites) are the way to do this. ([Here](https://vercel.com/docs/edge-network/rewrites) is a longer article)

**Reply by TheRealSalmon** [reply]
My problem was really dumb ... I needed to edit the GitHub Action to copy `vercel.json` into `dist`. I will open a PR on the example repo? This seems relevant to a tutorial repo deploying leptos on vercel.
